mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ExecutorStateManagerImpl.java

```diff
@@ -542,18 +546,32 @@ private Optional<TaskExecutorGroupKey> findBestGroup(SchedulingConstraints reque
      *         If no suitable key is found, it returns an empty Optional.
      */
     private Optional<TaskExecutorGroupKey> findBestGroupBySizeNameMatch(SchedulingConstraints requestedConstraints) {
+        // Filter to retain groups where sizeName is present
+        // Filter to retain groups where the requested sizeName is also present
+        // Filter to retain groups where sizeNames of group and requested constraints are equal
+        // Verify scheduling attribute constraints
+        // Sorted descending by generation
+        // Get highest generation group
         return executorsByGroup.keySet()
```

Comment:
I am a little lost here: are same SKU id under different generations sharing the same group key?